### PR TITLE
Replace swarm Capabilites API with cap add/drop API

### PR DESCRIPTION
**Pull request by cpuguy83**
Replace service "Capabilities" w/ add/drop API
    
After dicussing with maintainers, it was decided putting the burden of
providing the full cap list on the client is not a good design.
Instead we decided to follow along with the container API and use cap
add/drop.
    
This brings in the changes already merged into swarmkit.

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -3276,18 +3267,27 @@ definitions:
...
               type: "string"
             example:
               - "CAP_NET_RAW"
               - "CAP_SYS_ADMIN"
               - "CAP_SYS_CHROOT"
               - "CAP_SYSLOG"
+          CapabilityDrop:
+            type: "array"
+            description: |
+              A list of kernel capabilities to be drop from the default set
```
Comment (thaJeztah) [inline api/swagger.yaml:3285]:
```suggestion
              A list of kernel capabilities to drop from the default set
```

or

```suggestion
              A list of kernel capabilities to be dropped from the default set
```

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by tonistiigi — APPROVED** [review]

**Comment by olljanat** [comment]
SGTM

**Comment by thaJeztah** [comment]
Looks like compile is failing;

```

[2020-07-25T07:30:40.510Z] daemon/oci_windows.go:393:45: too many arguments in call to caps.TweakCapabilities
[2020-07-25T07:30:40.510Z] daemon/oci_windows.go:393:129: c.HostConfig.Capabilities undefined (type *"github.com/docker/docker/api/types/container".HostConfig has no field or method Capabilities)
```

File: docs/api/version-history.md [hunk 2]
```diff
@@ -28,12 +28,12 @@ keywords: "API, Docker, rcli, REST, documentation"
 * The `filter` (singular) query parameter, which was deprecated in favor of the
   `filters` option in Docker 1.13, has now been removed from the `GET /images/json`
   endpoint. The parameter remains available when using API version 1.40 or below.
-* `GET /services` now returns `Capabilities` as part of the `ContainerSpec`.
-* `GET /services/{id}` now returns `Capabilities` as part of the `ContainerSpec`.
-* `POST /services/create` now accepts `Capabilities` as part of the `ContainerSpec`.
-* `POST /services/{id}/update` now accepts `Capabilities` as part of the `ContainerSpec`.
-* `GET /tasks` now  returns `Capabilities` as part of the `ContainerSpec`.
-* `GET /tasks/{id}` now  returns `Capabilities` as part of the `ContainerSpec`.
+* `GET /services` now returns `CappAdd` and `CapDropp` as part of the `ContainerSpec`.
```
Comment (thaJeztah) [inline docs/api/version-history.md:31]:
```suggestion
* `GET /services` now returns `CappAdd` and `CapDrop` as part of the `ContainerSpec`.
```

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

(spotted one typo, but not a blocker)

File: docs/api/version-history.md (hunk 2 above)
Comment (cpuguy83) [reply docs/api/version-history.md:31]: Fixed

**Comment by thaJeztah** [comment]
Sigh

```
[2020-07-27T17:20:11.681Z] go: extracting gotest.tools/gotestsum v0.3.5
[2020-07-27T17:22:54.030Z] verifying gopkg.in/gemnasium/logrus-airbrake-hook.v2@v2.1.2/go.mod: gopkg.in/gemnasium/logrus-airbrake-hook.v2@v2.1.2/go.mod: Get https://sum.golang.org/lookup/gopkg.in/gemnasium/logrus-airbrake-hook.v2@v2.1.2: read tcp 172.21.183.185:49160->216.58.195.81:443: wsarecv: An existing connection was forcibly closed by the remote host.
[2020-07-27T17:22:54.030Z] verifying github.com/mattn/go-isatty@v0.0.3/go.mod: github.com/mattn/go-isatty@v0.0.3/go.mod: reading https://sum.golang.org/lookup/github.com/mattn/go-isatty@v0.0.3: read tcp 172.21.183.185:49160->216.58.195.81:443: wsarecv: An existing connection was forcibly closed by the remote host.
[2020-07-27T17:22:54.030Z] verifying github.com/stretchr/objx@v0.1.0/go.mod: github.com/stretchr/objx@v0.1.0/go.mod: Get https://sum.golang.org/lookup/github.com/stretchr/objx@v0.1.0: read tcp 172.21.183.185:49160->216.58.195.81:443: wsarecv: An existing connection was forcibly closed by the remote host.
```

**Comment by thaJeztah** [comment]
Looks like this one is flaky again

```
=== RUN   TestDockerSuite/TestLogsFollowSlowStdoutConsumer
    --- FAIL: TestDockerSuite/TestLogsFollowSlowStdoutConsumer (1.58s)
        docker_cli_logs_test.go:243: assertion failed: error is not nil: exit status 1

**Comment by thaJeztah** [comment]
All green now

**Comment by dmandalidis** [comment]
Placing a note here for anyone falling into this (like me). `HostConfig.Capabilities` (at least) were released with v1.40, therefore completely dropping them in favour of `capAdd/capDrop` didn't follow the usual deprecation process. Not a big deal, but I believe that this change was done under the impression that `HostConfig.Capabilities` were not released at all and their removal wouldn't break anyone. I guess this is also the reason that such change wasn't documented in the release notes of v1.41.

**Pull request merged**